src/consensus/aft/raft.h

```diff
@@ -2627,7 +2679,9 @@ namespace aft
 
       is_new_follower = true;
 
-      if (replica_state != kv::ReplicaState::Retired)
+      if (
+        replica_state != kv::ReplicaState::Retired &&
```

Comment:
I think we should have calls like takes_part_in_replication() and takes_part_in_election(), and the state conditions inside, for better readability.